pkg/objectupdate/rte/rte.go

```diff
@@ -191,6 +192,33 @@ func SecurityContext(ds *appsv1.DaemonSet, selinuxContextType string) {
 	}
 }
 
+type SecurityContextOptions struct {
+	SELinuxContextType  string
+	SecurityContextName string
+}
+
+func SecurityContextWithOpts(ds *appsv1.DaemonSet, opts SecurityContextOptions) {
+	cntSpec := objectupdate.FindContainerByName(ds.Spec.Template.Spec.Containers, manifests.ContainerNameRTE)
+	if cntSpec == nil {
+		return
+	}
+
+	// this is needed to put watches in the kubelet state dirs AND
+	// to open the podresources socket in R/W mode
+	if cntSpec.SecurityContext == nil {
+		cntSpec.SecurityContext = &corev1.SecurityContext{}
+	}
+	cntSpec.SecurityContext.SELinuxOptions = &corev1.SELinuxOptions{
+		Type:  opts.SELinuxContextType,
```

Comment:
this must be `!= ""` and the function should error out if it is nil (yes, this is stricter than the original function)